packages/mwp-app-render/src/util/dateTimeUtils.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Pads numbers less than two digits with an additional
+ * leading zero. Used for date and time formatting.
+ * @param {Number} num  Number to be padded
+ * @returns {String}    Returns a padded string
+ */
+export const padNumber = num => (num < 10 ? `0${num}` : `${num}`);
+
+/**
+ * Returns a formatted time string from the local value of the datetime
+ * object passed in for usage with the `find/upcoming_events` endpoint.
+ * The formatted date must follow the format: `HH:MM:SS`. Javascript's
+ * `.toISOString` method returns the iso value while we need the local value.
+ * @param {Object} timeObj  Datetime object
+ * @returns {String}        Formatted local date string in format `HH:MM:SS`
+ */
+export const getFormattedTime = timeObj =>
+        timeObj.toLocaleTimeString('UTC', { hour12: false });
+
+/**
+ * Returns a Date object that offsets the supplied UTC time, correcting for the runtime environment's current UTC offset.
```

Comment:
Given how confusing this behavior of this function is and the specific caveats for how it should be used, I feel like it needs a much more comprehensive description - what use cases does it apply to? Why would it be useful to manually apply 2 timezone shifts on a Date? How is daylight savings time accounted for?

Without docs, the best case scenario is that developers spend a lot of time discussing the implementation each time it is used - worst case, it is used in 'inappropriate' contexts and bugs appear.

Reply:
@jmhammons I tried a stab at some of this explanation here, if this helps at all: https://meetup.atlassian.net/wiki/spaces/INTL/pages/353338155/Engineering+SEO

@mmcgahan I think you have about the best understanding of anyone I've talked to about what this does

Reply:
maybe we should start a 'how to work with dates' document somewhere...

I'm fine for this code comment to be a confluence link, (although we only have weak guarantees of keeping those URLs consistent), so if anyone wants to start a page like that in the Web Eng Collective space, that would be great, but I'd like to avoid the necessarily-wandering discussions about when/how to use this in the future :)